ethereum/core/src/test/java/tech/pegasys/teku/core/signatures/DeletableSignerTest.java

```diff
@@ -0,0 +1,230 @@
+    when(delegate.signAggregationSlot(UInt64.ONE, forkInfo)).thenReturn(signatureFuture);
+    assertThatSafeFuture(signer.signAggregationSlot(UInt64.ONE, forkInfo))
+        .isCompletedWithValue(signature);
+  }
+
+  @Test
+  void signAggregationSlot_shouldNotSignWhenDisabled() {
+    signer.disable();
+    assertThatSafeFuture(signer.signAggregationSlot(UInt64.ONE, forkInfo))
+        .isCompletedExceptionallyWith(SignerNotActiveException.class);
+    verify(delegate, never()).signAggregationSlot(UInt64.ONE, forkInfo);
+  }
+
+  @Test
+  void signAggregateAndProof_shouldSignWhenActive() {
+    final AggregateAndProof aggregateAndProof = dataStructureUtil.randomAggregateAndProof();
+    when(delegate.signAggregateAndProof(aggregateAndProof, forkInfo)).thenReturn(signatureFuture);
+    assertThatSafeFuture(signer.signAggregateAndProof(aggregateAndProof, forkInfo))
+        .isCompletedWithValue(signature);
+  }
+
+  @Test
+  void signAggregateAndProof_shouldNotSignWhenDisabled() {
+    final AggregateAndProof aggregateAndProof = dataStructureUtil.randomAggregateAndProof();
+    signer.disable();
+    assertThatSafeFuture(signer.signAggregateAndProof(aggregateAndProof, forkInfo))
+        .isCompletedExceptionallyWith(SignerNotActiveException.class);
+    verify(delegate, never()).signAggregateAndProof(aggregateAndProof, forkInfo);
+  }
+
+  @Test
+  void signVoluntaryExit_shouldSignWhenActive() {
+    final VoluntaryExit aggregateAndProof = dataStructureUtil.randomVoluntaryExit();
+    when(delegate.signVoluntaryExit(aggregateAndProof, forkInfo)).thenReturn(signatureFuture);
+    assertThatSafeFuture(signer.signVoluntaryExit(aggregateAndProof, forkInfo))
+        .isCompletedWithValue(signature);
+  }
+
+  @Test
+  void signVoluntaryExit_shouldNotSignWhenDisabled() {
+    final VoluntaryExit aggregateAndProof = dataStructureUtil.randomVoluntaryExit();
+    signer.disable();
+    assertThatSafeFuture(signer.signVoluntaryExit(aggregateAndProof, forkInfo))
+        .isCompletedExceptionallyWith(SignerNotActiveException.class);
+    verify(delegate, never()).signVoluntaryExit(aggregateAndProof, forkInfo);
+  }
+
+  @Test
+  void signSyncCommitteeMessage_shouldSignWhenActive() {
+    final Bytes32 root = dataStructureUtil.randomBytes32();
+    when(delegate.signSyncCommitteeMessage(UInt64.ONE, root, forkInfo)).thenReturn(signatureFuture);
+    assertThatSafeFuture(signer.signSyncCommitteeMessage(UInt64.ONE, root, forkInfo))
+        .isCompletedWithValue(signature);
+  }
+
+  @Test
+  void signSyncCommitteeMessage_shouldNotSignWhenDisabled() {
+    final Bytes32 root = dataStructureUtil.randomBytes32();
+    signer.disable();
+    assertThatSafeFuture(signer.signSyncCommitteeMessage(UInt64.ONE, root, forkInfo))
+        .isCompletedExceptionallyWith(SignerNotActiveException.class);
+    verify(delegate, never()).signSyncCommitteeMessage(UInt64.ONE, root, forkInfo);
+  }
+
+  @Test
+  void signSyncCommitteeSelectionProof_shouldAlwaysSign() {
```

Comment:
shouldSignWhenActive?

Reply:
i had copied a lot of the names and clearly didnt clean them all up! thanks